spec/support/database_cleaner.rb

```diff
@@ -1,6 +1,16 @@
 RSpec.configure do |config|
-  config.before(:each) do
+  config.before(:suite) do
     DatabaseCleaner.clean_with(:truncation)
```

Comment:
> In fact, the ideal solution would be running the linting code in a task

Yes Indeed, my initial approach was the same but then I thought of doing it this way.
>And to be honest I don't know why the mock_commercial_request call is inside the if ENV['OSEM_FACTORY_LINT'] != 'false' as this is completely unrelated to the lint stuff.

I was in the same confusion but later find out that our factory is dependent on internet beacuse of this validation in which we check if the commercial is valid.
https://github.com/openSUSE/osem/blob/master/app/models/commercial.rb#L25
So we need to mock the request before linting the factories.